### PR TITLE
Add comment field to parameters

### PR DESCRIPTION
**Pull request by ntheanh201**
Closes: #7456 

Add field "comment" to parameter

## Which Traffic Control components are affected by this PR?
- Traffic Ops
- Traffic Portal

## What is the best way to verify this PR?
- Enable show parameters "comment" field in traffic_portal_properties.json: properties.parameters.comment.show = true
- Provisioning CDN in a Box, select Configure/Parameters, 1 more columns "comment" should show
- Create/Update Parameters with "comment"

## PR submission checklist
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Comment by ocket8888** [comment]
I am -1 on this idea. The promise of a comments field being used only for humans' benefits has been repeatedly broken on other objects, and a Parameter ought to be too small to express anything complex enough to require a comment. In fact, Parameters as objects should not even exist in their own right, IMO, and - especially now that Profile layering is possible - instead just be properties of Profiles. In some respects, they are represented that way already, and that means that this amounts to, in those cases, having an unbounded number of free-form text fields on a collection of free-form text fields on a Profile.

**Comment by ntheanh201** [comment]
> at way already, and that means that this amounts to, in those cases, having an unbounded number of free-form text fields on a collection of free-form text fields on a Profile.

IMO I still need a description of some special parameters that are required for the Profile

**Comment by ocket8888** [comment]
Any Parameter that is required for a Profile of some given type ought not to be optionally provided by the data model; instead it should be a required property of a Profile.

**Comment by mitchell852** [comment]
@ntheanh201 - can you make it optional to show this field in traffic portal? for example, add a config setting like parameter.comment.show = false to the [traffic portal properties file](https://github.com/apache/trafficcontrol/blob/master/traffic_portal/app/src/traffic_portal_properties.json) and key off that in the UI.

**Comment by ntheanh201** [comment]
> @ntheanh201 - can you make it optional to show this comment field in traffic portal? for example, add a config setting like parameter.comment.show = false to the [traffic portal properties file](https://github.com/apache/trafficcontrol/blob/master/traffic_portal/app/src/traffic_portal_properties.json) and key off that in the UI.

Updated like this, please take a look!

**Comment by ocket8888** [comment]
I'm still -1 on this idea. Adding tech debt doesn't seem like a good way to resolve tech debt.

**Review by kdamichie — CHANGES_REQUESTED** [review]

**Review by kdamichie — APPROVED** [review]
Looks good.  Comment field is present under parameters

**Comment by kdamichie** [comment]
@ntheanh201 ^

**Comment by ntheanh201** [comment]
> Sorry to amend my response. But can you verify this test failure. Looks like it needs a test update.

It has not failed because of this code, this failure is related to the integration tests themselves being broken, you can see some other commits

**Review by kdamichie — APPROVED** [review]
Approved

**Comment by zrhoffman** [comment]
@ocket8888 Responded in https://github.com/apache/trafficcontrol/issues/7456#issuecomment-1917912002 because a PR is not the place to discuss the validity of an Issue. With no response after a week, I'll assume you agree and will merge

**Comment by ntheanh201** [comment]
@zrhoffman can you merge?

**Comment by zrhoffman** [comment]
> @zrhoffman can you merge?

If we can verify that no committers are against #7845 being merged, yes. I'll clarify in Issue #7456 :)

File: traffic_ops/app/db/migrations/2023121910512261_add_comment_parameter.down.sql
Comment (zrhoffman) [inline traffic_ops/app/db/migrations/2023121910512261_add_comment_parameter.down.sql:1]: The timestamp on this migration, `2023121910512261`, is 10 months old. Would you please give it a more recent timestamp (for example, `2024101800000000`)?

File: traffic_ops/app/db/migrations/2023121910512261_add_comment_parameter.up.sql
Comment (zrhoffman) [inline traffic_ops/app/db/migrations/2023121910512261_add_comment_parameter.up.sql:1]: Same thing here, the timestamp `2023121910512261` is from last year and should be replaced with something more recent.

**Review by zrhoffman — CHANGES_REQUESTED** [review]
Just a small change needed with the migration timestamp, then it should be good to merge. Thanks!

**Comment by ntheanh201** [comment]
@zrhoffman I updated it

**Review by zrhoffman — APPROVED** [review]
Look great, thanks a lot! :)

**Pull request merged**
